components/tidb_query/src/codec/mysql/time/mod.rs

```diff
@@ -824,6 +815,32 @@ impl ConvertTo<Decimal> for Time {
     }
 }
 
+impl ConvertTo<Json> for Time {
+    #[inline]
+    fn convert(&self, _: &mut EvalContext) -> Result<Json> {
+        let s = if self.time_type == TimeType::DateTime || self.time_type == TimeType::Timestamp {
+            // TODO: avoid this clone
+            let mut val = self.clone();
+            val.fsp = mysql::MAX_FSP as u8;
+            val.to_string()
+        } else {
+            self.to_string()
+        };
+        Ok(Json::String(s))
+    }
+}
+
+impl ConvertTo<MyDuration> for Time {
+    fn convert(&self, _: &mut EvalContext) -> Result<MyDuration> {
+        if self.is_zero() {
+            return Ok(MyDuration::zero());
+        }
+        let nanos = i64::from(self.time.num_seconds_from_midnight()) * NANOS_PER_SEC
+            + i64::from(self.time.nanosecond());
+        MyDuration::from_nanos(nanos, self.fsp as i8)
+    }
```

Comment:
this is not same as 
```go
	hour, minute, second := t.Clock()
	frac := t.Time.Microsecond() * 1000

	d := gotime.Duration(hour*3600+minute*60+second)*gotime.Second + gotime.Duration(frac)
	// TODO: check convert validation
	return Duration{Duration: d, Fsp: t.Fsp}, nil
```

Reply:
They have the same result (`gotime.Duration` is an int64 value which represents in nano seconds).